gtda/diagrams/features.py

```diff
@@ -494,3 +498,208 @@ def transform(self, X, y=None):
 
         Xt = np.concatenate(Xt).reshape(self._n_dimensions, len(X)).T
         return Xt
+
+@adapt_fit_transform_docs
+class ATOL(BaseEstimator, TransformerMixin):
+    """Implementation of Automatic Topologically-Oriented Learning. Vectorises
+    persistence diagrams after a quantisation step.
+
+    Parameters
+    ----------
+    quantiser : ``'KMeans'`` or ``'MiniBatchKMeans'``, optional, default: \
+        ``'KMeans'``
+        Clustering algorithm used in the quantisation step.
+
+    quantiser_params : dict or list of n_dimensions dicts, optional, \
+        default: ``{'n_clusters': 10}``
+        Keyword argument for the quantiser. If it is a list, then each
+        dictionary corresponds to the keyword argument for the quantiser
+        applied to its corresponding homology dimension.
+
+    weight_function : ``'uniform'`` or None, optional, default: ``None``
+        Constant generic function for weighting the measure points. If
+        ``None``, all weights are set to 1.
+
+    contrast_function : ``'gaussian'`` | ``'laplacian'`` | ``'indicator'``, \
+        optional, default: ``'gaussian'``
+        Contrast function for evaluating proximity of a measure with respect
+        to centers.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1 unless
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    effective_quantiser_params_ : list of n_dimensions dict
+        List of dictionary containing all information present in
+        `quantiser_params` for each homology dimensions.
+
+    homology_dimensions_ : list
+        Homology dimensions seen in :meth:`fit`, sorted in ascending order.
+
+    centers_ : ndarray of shape (1, n_clusters, 3)
+        Diagram made of cluster centers for each homology dimensions.
```

Comment:
Suggestion: Location of cluster centers in each of the homology dimensions.

Also, the shape is something like `(1, sum(n_clusters), 3)`, isn't it?